cpp/src/strings/utilities.hpp

```diff
@@ -111,6 +112,34 @@ std::unique_ptr<column> make_empty_strings_column(
     rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource(),
     cudaStream_t stream = 0);
 
+/**
+ * @brief Utility to create a null mask for a strings column using a custom function.
+ *
+ * @tparam BoolFn Function should return true/false given an index for a strings column.
+ * @param strings_count Number of strings for the column.
+ * @param bfn The custom function used for identifying null string entries.
+ * @param mr Memory resource to use to device allocation.
+ * @param stream Stream to use for any kernel calls.
+ * @return The null mask and null count returned as std::pair.
+ */
+template <typename BoolFn>
+std::pair<rmm::device_buffer,cudf::size_type> make_null_mask( cudf::size_type strings_count,
```

Comment:
This function looks like it can be replaced entirely by https://github.com/rapidsai/cudf/pull/3263

Reply:
Not quite. The `valid_if` in #3263 always returns a null-mask even if all are valid (no nulls). This utility can be updated to call that one once the new one is ready.

Reply:
What's wrong with that? Doesn't seem worth maintaining a function that is 99% redundant for something like that.

Reply:
The `expect_columns_equal` and specifically the `expect_column_properties_equal` methods expect `nullable()` to match in both columns. https://github.com/rapidsai/cudf/blob/f601939122ea979d15b90e293018616703876e3f/cpp/tests/utilities/column_utilities.cu#L37
This will fail if one column is created with all-valid null mask and the other is created with no null mask even though both are technically equivalent.